contracts/Fund.sol

```diff
@@ -538,49 +549,52 @@ contract Fund is ERC20, DestructiblePausable {
   // 1) transfer and transferFrom check that the recipient is eligible based on their allocation
   // 2) the sharesOwned variable in the Investor struct is identical to balances
 
-  mapping(address => uint256) balances;
-  mapping (address => mapping (address => uint256)) allowed;
+  mapping(address => uint) balances;
+  mapping (address => mapping (address => uint)) allowed;
 
-  function transfer(address _to, uint256 _value)
+  function transfer(address _to, uint _value)
     whenNotPaused
-    returns (bool)
+    returns (bool success)
   {
-    if (investorActions.checkEligibility(_to, _value)) {
-      investors[msg.sender].sharesOwned = investors[msg.sender].sharesOwned.sub(_value);
-      balances[msg.sender] = balances[msg.sender].sub(_value);
-      investors[_to].sharesOwned = investors[_to].sharesOwned.add(_value);
-      balances[_to] = balances[_to].add(_value);
-      Transfer(msg.sender, _to, _value);
-      return true;
-    } else {
-      return false;
-    }
+    require(_to != address(0));
+    require(_value <= balances[msg.sender]);
+    require(_value <= investors[msg.sender].sharesOwned);
+    require(_value <= toShares(investorActions.getAvailableAllocation(_to)));
```

Comment:
If `getAvailableAllocation` provides the remaining allocation available to an investor, does this check make sense?  

For instance, if I have total allocation of 10 ETH and I fill it, then would my `getAvailableAllocation` be zero?  If so, I should still be able to transfer the entire 10 ETH.

Reply:
Wait, nevermind - I just realized that you are checking the recipient's allocation.  That makes sense.